tpm2/structures.go

```diff
@@ -371,3 +371,158 @@ type tpmtSigScheme struct {
 	Scheme Algorithm
 	Hash   Algorithm
 }
+
+// AttestationData contains data attested by TPM commands (like Certify).
+type AttestationData struct {
+	Magic               uint32
+	Type                tpmutil.Tag
+	QualifiedSigner     Name
+	ExtraData           []byte
+	ClockInfo           ClockInfo
+	FirmwareVersion     uint64
+	AttestedCertifyInfo *CertifyInfo
+}
+
+// DecodeAttestationData decode a TPMS_ATTEST message. No error is returned if
+// in has extra trailing data.
+func DecodeAttestationData(in []byte) (*AttestationData, error) {
+	buf := bytes.NewBuffer(in)
+
+	var ad AttestationData
+	if err := tpmutil.UnpackBuf(buf,
```

Comment:
style: this doesn't seem consistent to me with other code so far: normally, UnpackBuf has been on a single line. I find this compound if statement hard to read like this. Can this (and similar lines) be pulled back up into a single line? If the line gets really long, then I don't mind breaking it up, but in this case, it's

```go
if err := tpmutil.UnpackBuf(buf, &ad.Magic, &ad.Type); err != nil {
```
and that seems reasonable; it even fits on an 80-char line.

Reply:
Done